packages/router/src/util.ts

```diff
@@ -67,11 +69,27 @@ type SupportedRouterParamTypes = keyof typeof coreParamTypes
  *  matchPath('/post/{id:Int}', '/post/7')
  *  => { match: true, params: { id: 7 }}
  */
+
+export const formatPath = (
+  path: string,
+  trailingSlashes: TrailingSlashesTypes
+) => {
+  return {
+    never: path.replace(/\/$/, ''),
+    always: path.endsWith('/') ? path : path + '/',
+    preserve: path,
+  }[trailingSlashes]
+}
+
 const matchPath = (
-  route: string,
-  pathname: string,
-  paramTypes?: Record<string, ParamType>
+  path: string,
+  locationPathname: string,
+  paramTypes?: Record<string, ParamType>,
+  trailingSlashes: TrailingSlashesTypes = 'never'
 ) => {
+  const route = formatPath(path, trailingSlashes)
+  const pathname = formatPath(locationPathname, trailingSlashes)
+
```

Comment:
`never` and `always` are essentially the same thing one adds and one removes the slashes. But it is a little awkward.

I may have messed up the implementation a little bit. And would like a little guidance. 
The expected result is to match both `/post` and `/post/` according to the issue

Should we be manipulating the `location.pathname` here as well for matches?

```
never - Always strip trailing slashes from URLs before matching.
always - Always add trailing slashes to URLs before matching.
```

The way I read it and implemented it strips/adds the slashes for all the URLs involved aka the `route` and `location.pathname`

Reply:
Super late feedback (but I'm keen to get this merged soon!)—I think I get the conundrum you're facing now. From reading the proposal, it sounds like we should only be manipulating the `location.pathname`, not the `path`. That terminology still doesn't make things super clear, so basically, given this route:

```js
<Route path="/post" page={PostPage} name="post" />
```

where the path is `/post`, the `trailingSlash` policy `never` makes the most sense (the path has no trailing slash, so we probably want to strip it from the URL). But if the user specified `always`, this route wouldn't match anything. So it kind of makes sense to add the slash to the path and the URL if the policy is `always`, so that things work. But I think we shouldn't be changing the path as it's specified in `Routes.js` at all. So then the question is, do we just let things kind of fail? Or we do provide a linting error at least to say that "this path isn't matching anything".

I'll think it over and double check with the team!